### PR TITLE
Development

### PR DESCRIPTION
**Pull request by Joyan11**
i would really appreciate a code review on this.
Do let me know where i can improve and if there's a better way to do something

Note : class is used in ionicons tag because className dosent work for this set of icon provider in react

### CODE REVIEW

**Review by tuhindas30 — COMMENTED** [review]
Hey @Joyan11,
I had gone through your code. It is greatly written. I loved the way how you made apis for each operation. Each and every line looks neat and clean. However, some very small nitty changes that I would suggest and it's okay to keep it like this also.
* `faker` and `react-uuid` packages can be removed as the backend is involved.
* I think the `manifest.json` file is of no use. It can also be removed.
Once again this looks great. I learned a lot from your code. Keep going!

**Comment by shraddhaa98** [comment]
Hi Joyan,

Great work on this project! Neat code. I liked how modular your code base is. Most core functionalities have been broken down into separate components. One suggestion would be to include an index file in the components folder. You can export all the components into this file. This way when you're trying to use the components, you can import them from a single location. This would reduce the number of import statements and make it easier to understand.

**Comment by Joyan11** [comment]
> Hi Joyan,
> 
> Great work on this project! Neat code. I liked how modular your code base is. Most core functionalities have been broken down into separate components. One suggestion would be to include an index file in the components folder. You can export all the components into this file. This way when you're trying to use the components, you can import them from a single location. This would reduce the number of import statements and make it easier to understand.

hello,
Thank you very much for the review. I made the following changes in the code. this was a very useful suggestion , imports look a lot more organized now

**Comment by Joyan11** [comment]
> Hey @Joyan11,
> I had gone through your code. It is greatly written. I loved the way how you made apis for each operation. Each and every line looks neat and clean. However, some very small nitty changes that I would suggest and it's okay to keep it like this also.
> 
> * `faker` and `react-uuid` packages can be removed as the backend is involved.
> * I think the `manifest.json` file is of no use. It can also be removed.
>   Once again this looks great. I learned a lot from your code. Keep going!

Hello, 
Thanks for the review. I've removed the unwanted packages. Cheers!

File: src/App.js [hunk 1]
```diff
@@ -23,16 +32,21 @@ export default function App() {
...
+          wishList={wishList}
         />
         <Routes>
           <Route path="/" element={<Home />} />
           <Route path="/products" element={<Products />} />
-          <Route path="/cart" element={<Cart />} />
-          <Route path="/wishList" element={<Wishlist />} />
           <Route path="/products/:id" element={<ProductDetail />} />
+          <Route path="/login" element={<Login />} />
+          <Route path="/signup" element={<Signup />}></Route>
```
Comment (renukapaturkar) [inline src/App.js:58]: Keep the`<Route></Route>` as `<Route/>`

File: src/api/cart/removeFromCart.js [hunk 2]
```diff
@@ -1,19 +1,20 @@
...
-      status,
-      // data: {
-      //   cartItems: { _id: cartid, products },
-      // },
-    } = await axios.delete(
-      `https://rabonaserver.joyan11.repl.co/cart/${cartId}/${productid}`
+    dispatch({ type: "CART_ACTION_LOADER", payload: productid });
+    const { status } = await axios.delete(
+      `https://rabonaserver.joyan11.repl.co/cart/${cartId}/${productid}`,
+      { headers: { authorization: token } }
```
Comment (renukapaturkar) [inline src/api/cart/removeFromCart.js]: Try to put the common authorization header for all in a single function or  you can use axios interceptors. 
To know more about this check [here](https://axios-http.com/docs/config_defaults)

**Review by renukapaturkar — CHANGES_REQUESTED** [review]
Hi Joyan, 
Your code is very well written and files are properly structured and modular. I have mentioned a small change to optimize code and avoid repetition and please make sure to remove all the `console.log()` and commented code once you are development is done.

I have leant a lot from this code review, Thank you.

**Pull request closed without merging**